packages/core/crates/core-crypto/src/derivation.rs

```diff
@@ -127,6 +132,104 @@ pub fn derive_ack_key_share(secret: &[u8]) -> HalfKey {
     sample_field_element(secret, HASH_KEY_ACK_KEY).expect("failed to sample ack key share")
 }
 
+/// Bundles values given to the smart contract to prove that a ticket is a win.
+///
+/// The VRF is thereby needed because it generates on-demand determinitstic
+/// entropy that can only be derived by the ticket redeemer.
+pub struct VrfParameters {
+    /// the pseudo-random point
+    pub v: AffinePoint,
+    pub h: Scalar,
+    pub s: Scalar,
+    /// helper value for smart contract
+    pub h_v: AffinePoint,
+    /// helper value for smart contract
+    pub s_b: AffinePoint,
+}
+
+impl std::fmt::Display for VrfParameters {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let v_encoded = self.v.to_encoded_point(false);
+        let h_v_encoded = self.h_v.to_encoded_point(false);
+        let s_b_encoded = self.s_b.to_encoded_point(false);
+        f.debug_struct("VrfParameters")
+            .field(
+                "V",
+                &format!(
+                    "({},{})",
+                    hex::encode(v_encoded.x().unwrap()),
+                    hex::encode(v_encoded.y().unwrap())
+                ),
+            )
+            .field("h", &hex::encode(self.h.to_bytes()))
+            .field("s", &hex::encode(self.s.to_bytes()))
+            .field(
+                "h_v",
+                &format!(
+                    "({},{})",
+                    hex::encode(h_v_encoded.x().unwrap()),
+                    hex::encode(h_v_encoded.y().unwrap())
+                ),
+            )
+            .field(
+                "s_b",
+                &format!(
+                    "({},{})",
+                    hex::encode(s_b_encoded.x().unwrap()),
+                    hex::encode(s_b_encoded.y().unwrap())
+                ),
+            )
+            .finish()
+    }
+}
+
+/// Takes a private key, the corresponding Ethereum address and a payload
+/// and creates all parameters that are required by the smart contract
+/// to prove that a ticket is a win.
+pub fn derive_vrf_parameters<const T: usize>(
+    msg: &[u8; T],
+    secret: &[u8],
+    chain_addr: &Address,
+    dst: &[u8],
+) -> Result<VrfParameters> {
+    let b = Secp256k1::hash_from_bytes::<ExpandMsgXmd<sha3::Keccak256>>(&[&chain_addr.to_bytes(), msg], &[dst])?;
+
+    let a: Scalar = ScalarPrimitive::<Secp256k1>::from_slice(&secret)?.into();
```

Comment:
I'd rather use `NonZeroScalar::<Secp256k1>::try_from(secret)` here

Reply:
@robertkiel did a different approach by checking if the scalar is non-zero in an added condition. That should be equivalent.